File: app/src/at-tamle/java/asiantech/internship/winter/baitap/bai28.java
package asiantech.internship.winter.baitap;
import java.util.*;
//Viết chương trình thực hiện chuẩn hoá một xâu ký tự nhập từ bàn phím
//loại bỏ các dấu cách thừa, chuyển ký tự đầu mỗi từ thành chữ hoa, các ký tự khác thành chữ thường)
public class bai28 {
	
	
	 public static String chuyenInHoa(String str){ 
	                String s,b;
	                s= str.substring(0, 1);
	 b= str.replaceFirst(s,s.toUpperCase()); 
	                return (b);
	        }
	 public static String chuanHoa(String strInput){ 
	                String b="";
	                StringTokenizer strToken= new StringTokenizer(strInput," ,\t,\r"); 
	                b+=""+chuyenInHoa(strToken.nextToken()); while(strToken.hasMoreTokens()){
	                   b+=" "+chuyenInHoa(strToken.nextToken());
	                }
	               return(b);
	       }
	       public static void main(String[] args) {
	 
	 Scanner input= new Scanner(System.in); 
	 System.out.println("Nhap vao 1 xau: "); 
	            String strInput= input.nextLine();
	            System.out.println("Xau duoc chuan hoa la: "+chuanHoa(strInput));
	      } 
	} 

File: app/src/at-trinhnguyen/java/asiantech/internship/winter/javacore/Bai28.kt
package asiantech.internship.winter.javacore

import java.util.*
import java.util.StringTokenizer

/*
Bài 28. Viết chương trình thực hiện chuẩn hoá một xâu ký tự nhập từ bàn phím (
loại bỏ các dấu cách thừa, chuyển ký tự đầu mỗi từ thành chữ hoa, các ký tự khác thành chữ thường)
 */
object Bai28 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        print("Enter string: ")
        val str = scanner.nextLine()
        println(standardized(str))
    }

    private fun firstCharToUpperCase(str: String): String {
        val s: String = str.substring(0, 1)
        val strOutput: String
        strOutput = str.replaceFirst(s.toRegex(), s.toUpperCase())
        return strOutput
    }

    private fun standardized(strInput: String): String {
        var strOutput = ""
        val strToken = StringTokenizer(strInput, " ,\t,\r")
        strOutput += "" + firstCharToUpperCase(strToken.nextToken())
        while (strToken.hasMoreTokens()) {
            strOutput += " " + firstCharToUpperCase(strToken.nextToken())
        }
        return strOutput
    }
}